CRM/Utils/Check/Component/Source.php

```diff
@@ -64,4 +67,25 @@ public function checkOrphans() {
     return $messages;
   }
 
+  /**
+   * Linux is case sensitive, so this will be a no-op.
+   * Windows is case insensitive, Mac is usually insensitive but sometimes
+   * sensitive.
+   * Note that realpath() will return the real casing for a file on windows,
+   * but not on mac, so we need a different method. glob returns the real
+   * casing, but means we need to loop.
+   *
+   * @param string $path
+   * @return string|null
+   */
+  private function findCorrectCaseForFile(string $path): ?string {
+    $fileToFind = basename($path);
+    foreach (glob(dirname($path) . '/*', GLOB_NOSORT) as $theRealFile) {
```

Comment:
Oh, nice. This glob trick does seem to work as a portable way to find out the effective capitalization. (At least, it works for me too on macOS.)

Reply:
It's just obviously slower. For the record I did try just `glob($path)` but it always returned the same case as what's in $path instead of the real case, at least on windows.